csup/metadata.go

```diff
@@ -150,6 +150,38 @@ func (f *Float) Len(*Context) uint32 {
 	return f.Count
 }
 
+type BytesTable struct {
+	Bytes   Segment
+	Offsets Segment
+	Count   uint32
+}
+
+type Bytes struct {
+	BytesTable BytesTable
+	Min, Max   []byte
+}
+
+func (b *Bytes) Type(*Context, *super.Context) super.Type {
+	return super.TypeString
+}
+
+func (b *Bytes) Len(*Context) uint32 {
+	return b.BytesTable.Count
+}
+
+type String struct {
+	BytesTable BytesTable
+	Min, Max   string
+}
+
+func (s *String) Type(*Context, *super.Context) super.Type {
+	return super.TypeString
+}
+
+func (s *String) Len(*Context) uint32 {
+	return s.BytesTable.Count
+}
+
```

Comment:
You could add just a single type with a super.Type field here. That'd let you eliminate duplicate code in csup/bytes.go too.

```suggestion
type Bytes struct {
	Typ     super.Type `super:"Type"`
	Bytes   Segment
	Offsets Segment
	Min     []byte
	Max     []byte
	Count   uint32
}

func (b *Bytes) Type(*Context, *super.Context) super.Type {
	return b.Typ
}

func (b *Bytes) Len(*Context) uint32 {
	return b.Count
}

Reply:
Great idea